lib/table.c

```diff
@@ -2304,39 +2315,36 @@ simplifier_merge_ancestors(simplifier_t *self, node_id_t input_id)
                     goto out;
                 }
             }
-            v = self->node_id_map[input_id];
-            if (v >= 0 && self->is_sample[input_id]) {
+            if (sample_node) {
                 /* If we have a mapped node over an interval with no other segments,
                  * then we must record this edge as it joins internal samples */
-                ret = simplifier_record_edge(self, alpha->left, alpha->right, v,
+                ret = simplifier_record_edge(self, alpha->left, alpha->right, output_id,
                         alpha->node);
                 if (ret != 0) {
                     goto out;
                 }
                 /* The node on this segment must be remapped to the parent so that we
                  * have the correct nodes further up in the tree. */
-                alpha->node = v;
+                alpha->node = output_id;
             }
         } else {
             if (!coalescence) {
```

Comment:
whereas this bit we want to do if the node is a sample **or** we've assigned this ancestor an `output_id` already because we found a coalescence here

Reply:
This is the bit I don't agree with. Why would we record a node if it's a sample? We've already done this in the sample case, haven't we? The coalescence flag is  for the case where a coalescence has happened, and we want to **avoid** recording a node mapping multiple times.

Reply:
Sorry, I meant to say that "this bit we **DO NOT** want to do if the node is a sample **or** we've already assigned an ancestor due to previously finding a coalescence." =\